src/ucp/core/ucp_ep.c

```diff
@@ -1393,13 +1408,15 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
                                      UCT_IFACE_FLAG_TAG_EAGER_ZCOPY, 0,
                                      iface_attr->cap.tag.eager.max_bcopy);
 
-            config->tag.offload.max_rndv_iov    = iface_attr->cap.tag.rndv.max_iov;
-            config->tag.offload.max_rndv_zcopy  = iface_attr->cap.tag.rndv.max_zcopy;
-            config->tag.sync_proto              = &ucp_tag_offload_sync_proto;
-            config->tag.proto                   = &ucp_tag_offload_proto;
-            config->tag.lane                    = lane;
-            max_rndv_thresh                     = iface_attr->cap.tag.eager.max_zcopy;
-            max_am_rndv_thresh                  = iface_attr->cap.tag.eager.max_bcopy;
+            config->tag.offload.max_rndv_iov   = iface_attr->cap.tag.rndv.max_iov;
+            config->tag.offload.max_rndv_zcopy = iface_attr->cap.tag.rndv.max_zcopy;
+            config->tag.sync_proto             = &ucp_tag_offload_sync_proto;
+            config->tag.proto                  = &ucp_tag_offload_proto;
+            config->tag.lane                   = lane;
+            max_rndv_thresh                    = iface_attr->cap.tag.eager.max_zcopy;
+            max_am_rndv_thresh                 = iface_attr->cap.tag.eager.max_bcopy;
+            min_rndv_thresh                    = ucp_ep_tag_offload_min_rndv_thresh(config);
```

Comment:
Should we calculate minimum between tag offload rndv thresh and iface’s GET Minimum Zcopy? Should do it in the new function?

Reply:
yes, but we need to take max between them

Reply:
Yes, it should be maximum between them
Thanks!

Btw, your code also should work since it will fallback to PUT or SW RNDV if message size lower than GET minimum Zcopy

Reply:
right, probably better not to change that then (to not complicate it further).